src/utils/fs.rs

```diff
@@ -209,3 +209,23 @@ pub fn try_infer_extension(path: &Path) -> Option<Extension> {
         None
     }
 }
+
+/// Copy the src directory into the dst directory recursively
+pub fn copy_dir(src: &Path, dst: &Path) -> crate::Result<()> {
+    if !src.exists() || !dst.exists() {
+        return Err(crate::Error::NotFound {
+            error_title: "source or destination directory does not exist".to_string(),
+        });
+    }
+
+    for entry in fs::read_dir(src)? {
+        let entry = entry?;
+        let ty = entry.file_type()?;
+        if ty.is_dir() {
+            copy_dir(entry.path().as_path(), dst.join(entry.file_name()).as_path())?;
+        } else {
+            fs::copy(entry.path(), dst.join(entry.file_name()).as_path())?;
```

Comment:
The only problem I see here is, this is duplicating everything that's already written to the disk, so an archive of 20GB can't be decompressed in a filesystem with only 15GB of free space (unless FS copy-on-write is enabled).

Instead of `cp` we should do a `mv`.

Can you change from `fs::copy` to `fs::rename` to achieve that?

(There is a caveat that this doesn't work well for crossing filesystem boundaries but that's a problem for future me.)